app-text/hunspell/hunspell-1.7.2-r1.ebuild

```diff
@@ -26,7 +26,7 @@ BDEPEND="
 	>=sys-devel/gettext-0.18
 "
 
-PDEPEND=""
+PDEPEND=" "
```

Comment:
sorry to ask, but why this change? (adding a whitespace)

Reply:
I see, well in that case `PDEPEND=""` could be removed alltogether.
However, there might be some setting in a eclass which initially sets `PDEPEND` (and gets emptied by this line). - in this case it might be a false-positive from `pkgcheck`.

Personally i guess it can be removed, but i haven't looked into it.
In any case setting it to `PDEPEND=" "` is certainly not the way to fix this.

Reply:
> I did not think that `+=`ing a non existing variable would work. I have run simple tests and it does. I have removed this line in [2340107](https://github.com/gentoo/gentoo/commit/2340107a2a9d4c84b4e7936361e88b6975c9be2b)

Sorry, but i'm not sure what you meant by that ;)
Thx for changing but you just commented the variable out. You could remove it completely. Also, the same applies to the second ebulid.

Reply:
@arthurzam I wonder if EmptyGlobalAssignment makes sense in an ebuild that has a later `+=` assignment, anyway?

Reply:
@eli-schwartz Again, why would you set to empty string in the first state? Is there any gain from that?

(I can update the logic to exclude such cases without issues, just trying to understand the case)

Reply:
One could argue that the purpose of setting it to an empty string is for visual purposes, to make it obvious that it's part of a loop that will append to it.

Conversely, the check by default is warning you for cases where people have logic to define variables to their default values, based on the misunderstanding that it is "required to define the variables, even if they are empty". Or at least that was my understanding.

pkgcheck isn't a bash linter -- it's an ebuild linter. The warning isn't telling you:
>  bash permits appending to nonexistent variables, no need to predeclare it

The warning is telling you:
> actually, portage does the right thing if, after sourcing the ebuild, a variable is unset instead of being set to ""